corehq/apps/custom_data_fields/edit_model.py

```diff
@@ -19,6 +19,7 @@ class CustomDataFieldsForm(forms.Form):
     The main form for editing a custom data definition
     """
     data_fields = forms.CharField(widget=forms.HiddenInput)
+    purge_existing = forms.BooleanField(widget=forms.HiddenInput, required=False, initial=False)
```

Comment:
hahaha it did take me a long time to figure out how some pieces worked, though the ability to be generically dropped in in multiple places is nice. i did at first try to reconcile the logic that handles the removal in the edit entity form with this, but that was too coupled to the view / request to be able to do it.